src/tensorwaves/interfaces.py

```diff
@@ -111,27 +119,56 @@ class Estimator(ABC):
     """Estimator for discrepancy model and data."""
 
     @abstractmethod
-    def __call__(
-        self, parameters: Mapping[str, Union[float, complex]]
-    ) -> float:
+    def __call__(self, parameters: Mapping[str, ParameterValue]) -> float:
         """Evaluate discrepancy."""
 
     @abstractmethod
     def gradient(
-        self, parameters: Mapping[str, Union[float, complex]]
-    ) -> Dict[str, Union[float, complex]]:
+        self, parameters: Mapping[str, ParameterValue]
+    ) -> Dict[str, ParameterValue]:
         """Calculate gradient for given parameter mapping."""
 
 
+@attr.s(frozen=True, auto_attribs=True)
+class FitResult:  # pylint: disable=too-many-instance-attributes
```

Comment:
I hesitated to put the `FitResult` under `interfaces`, because it's not an abstract class. Alternative would be to put it under some new sub-module like `tensorwaves.optimizer.result` (sub-module to avoid cyclic imports in `minuit` and `scipy`.

Reply:
I think its fine there, since its part of the Optimizer interface